ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogEntryIdComputationStrategy.scala

```diff
@@ -0,0 +1,70 @@
+package com.daml.ledger.validator
+
+import java.security.MessageDigest
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlLogEntryId,
+  DamlStateKey,
+  DamlSubmission,
+}
+import com.daml.ledger.participant.state.kvutils.Raw
+import com.daml.ledger.validator.LogEntryIdComputationStrategy._
+import com.google.protobuf.ByteString
+
+import scala.jdk.CollectionConverters._
+
+trait LogEntryIdComputationStrategy {
+  def compute(value: Raw.Envelope, submission: DamlSubmission): DamlLogEntryId
+}
+
+object LogEntryIdComputationStrategy {
+  val LogEntryIdPrefix = "0"
+
+  def hash(bytes: Array[Byte]): ByteString = {
+    val messageDigest = MessageDigest
+      .getInstance("SHA-256")
+    messageDigest.update(bytes)
+    val hash = messageDigest
+      .digest()
+      .map("%02x" format _)
+      .mkString
+    ByteString.copyFromUtf8(LogEntryIdPrefix + hash)
+  }
+}
+
+object RawLogEntryIdComputationStrategy extends LogEntryIdComputationStrategy {
+
+  // While the log entry ID is no longer the basis for deriving absolute contract IDs,
+  // it is used for keying log entries / fragments. We may want to consider content addressing
+  // instead and remove the whole concept of log entry identifiers.
+  // For now this implementation uses a sha256 hash of the submission envelope in order to generate
+  // deterministic log entry IDs.
+  override def compute(value: Raw.Envelope, submission: DamlSubmission): DamlLogEntryId =
+    DamlLogEntryId.newBuilder
+      .setEntryId(hash(value.bytes.toByteArray))
+      .build
+}
+
+object NoInputSubmissionLogEntryIdComputationStrategy extends LogEntryIdComputationStrategy {
+  override def compute(value: Raw.Envelope, submission: DamlSubmission): DamlLogEntryId = {
+    val newInputList = submission.getInputDamlStateList.asScala
+      .filter(_.getKeyCase match {
+        case DamlStateKey.KeyCase.CONTRACT_ID => false
```

Comment:
Note: this doesn't make too much sense, as I'm still trying to figure out the correct set of inputs for calculating the entry id.